pkg/util/storage/manager.go

```diff
@@ -82,3 +86,41 @@ func (m *manager) BlobService(ctx context.Context, resourceGroup, account string
 
 	return &blobcli, nil
 }
+
+func (m *manager) CreateBlobContainer(ctx context.Context, resourceGroup string, account string, container string) error {
+	needToCreateBlobContainer := false
+	_, err := m.blobContainers.Get(
+		ctx,
+		resourceGroup,
+		account,
+		container)
+	if err != nil {
+		if detailedErr, ok := err.(autorest.DetailedError); !ok || detailedErr.StatusCode != http.StatusNotFound {
+			return err
+		}
+		needToCreateBlobContainer = true
+	}
+
+	if !needToCreateBlobContainer {
+		return nil
+	}
+
+	_, err = m.blobContainers.Create(
+		context.Background(),
+		resourceGroup,
+		account,
+		container,
+		mgmtstorage.BlobContainer{
+			ContainerProperties: &mgmtstorage.ContainerProperties{
+				PublicAccess: mgmtstorage.PublicAccessNone,
```

Comment:
I believe we need to keep public access open if development mode is set, because we won't have an AFD instance. Otherwise, we should keep `PublicAccessNone`.

Reply:
I tried testing it in the local dev rp and it worked, maybe because I used the localFPAuthorizer, but I do have this access noted that it has to tested and be applied as per the environments. Starting with the lowest access available.
Thanks for noting this.

Reply:
This has been taken care of and it will always be `PublicAccessNone` for non-dev, otherwise `PublicAccessNone`. It can only be tested once we create the MIWI cluster after all the changes ready.